frontend/packages/client/src/contexts/NotificationService.js

```diff
@@ -84,54 +66,37 @@ const NotificationServiceProvider = ({ children }) => {
 
   const getUserSettings = async () => {
     try {
-      //here we call api
-      const { communitySubscription, isSubscribedFromCommunityUpdates } =
-        INIT_NOTIFICATION_SETTINGS;
+      const { communitySubscription, isSubscribedFromCommunityUpdates, email } =
+        await getUser(addr);
       setNotificationSettings((prevState) => ({
         ...prevState,
         communitySubscription,
         isSubscribedFromCommunityUpdates,
+        email,
       }));
     } catch {
       throw new Error('cannot get user settings');
     }
   };
 
-  const updateCommunitySubscription = async (
-    communityId,
-    subscribeIntention
-  ) => {
+  const updateCommunitySubscription = async (communitySubIntentions) => {
     try {
-      if (subscribeIntention === subscribeNotificationIntentions.subscribe) {
-        //call api to subscribe community
-      } else if (
-        subscribeIntention === subscribeNotificationIntentions.unsubscribe
-      ) {
-        //call api to unsubscribe community
-      }
-      setNotificationSettings((prevState) => {
-        const newCommunitySubscription = updateCommunitySubscriptionState(
-          prevState.communitySubscription,
-          communityId,
-          subscribeIntention === subscribeNotificationIntentions.subscribe
-        );
-        return {
-          ...prevState,
-          communitySubscription: newCommunitySubscription,
-        };
-      });
+      await updateCommunity(communitySubIntentions);
+      await new Promise((r) => setTimeout(r, 500));
```

Comment:
This is to throttle the api call, allow the data to get registered with leanplum before calling it again to get the updated data